src/main/java/seedu/address/logic/paynow/QrGenerator.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.logic.paynow;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Hashtable;
+import javax.imageio.ImageIO;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.EncodeHintType;
+import com.google.zxing.WriterException;
+import com.google.zxing.common.BitMatrix;
+import com.google.zxing.qrcode.QRCodeWriter;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+
+/**
+ * This helper class helps to generate a QR Code image given a string.
+ */
+public class QrGenerator {
+    private static final int SIZE = 400;
+    private static final Color PAYNOW_COLOR = Color.decode("#7a1b78");
+    private static final String LOGO_PATH = Paths.get(
+            "src", "main", "resources", "images", "paynowlogo.png").toString();
+
+    private static void overlayLogo(Graphics2D graphics) throws IOException {
+        Image logo = getLogo();
+        int startHeight = (SIZE - logo.getHeight(null)) / 2;
+        int startWidth = (SIZE - logo.getWidth(null)) / 2;
+        graphics.drawImage(logo, startWidth, startHeight, null);
+    }
+
+    private static Image getLogo() throws IOException {
+        BufferedImage originalLogo = ImageIO.read(new File(LOGO_PATH));
+        float scale = (float) SIZE / 6 / originalLogo.getHeight();
```

Comment:
Is 6 here considered a magic number? would it be better if it was somehow defined above as a static final variable?